src/third/redux-persist/getStoredState.js

```diff
@@ -18,27 +18,30 @@ export default function getStoredState (config, onComplete) {
   let restoredState = {}
   let completionCount = 0
 
-  storage.getAllKeys((err, allKeys) => {
-    if (err) {
+  storage.getAllKeys()
+    .catch(err => {
       if (process.env.NODE_ENV !== 'production') console.warn('redux-persist/getStoredState: Error in storage.getAllKeys')
       complete(err)
-      return
-    }
-
-    let persistKeys = allKeys.filter((key) => key.indexOf(keyPrefix) === 0).map((key) => key.slice(keyPrefix.length))
-    let keysToRestore = persistKeys.filter(passWhitelistBlacklist)
+    })
+    .then(allKeys => {
+      let persistKeys = allKeys.filter((key) => key.indexOf(keyPrefix) === 0).map((key) => key.slice(keyPrefix.length))
+      let keysToRestore = persistKeys.filter(passWhitelistBlacklist)
 
-    let restoreCount = keysToRestore.length
-    if (restoreCount === 0) complete(null, restoredState)
-    keysToRestore.forEach((key) => {
-      storage.getItem(createStorageKey(key), (err, serialized) => {
-        if (err && process.env.NODE_ENV !== 'production') console.warn('redux-persist/getStoredState: Error restoring data for key:', key, err)
-        else restoredState[key] = rehydrate(key, serialized)
-        completionCount += 1
-        if (completionCount === restoreCount) complete(null, restoredState)
+      let restoreCount = keysToRestore.length
+      if (restoreCount === 0) complete(null, restoredState)
+      keysToRestore.forEach(async (key) => {
+        try {
+          const serialized = await storage.getItem(createStorageKey(key))
+          // `rehydrate` won't throw an error; see implementation
+          restoredState[key] = rehydrate(key, serialized)
+        } catch (err) {
+          if (process.env.NODE_ENV !== 'production') console.warn('redux-persist/getStoredState: Error restoring data for key:', key, err)
+        } finally {
```

Comment:
This `finally` isn't doing anything -- with a `catch` that doesn't throw (and with no `return` or `continue` or `break` involved), we'll always proceed to the code after the whole try/catch statement anyway.